src/Fantomas.Tests/CommentTests.fs

```diff
@@ -1252,3 +1252,62 @@ let a = 8
 let a = 8
 // foobar
 """
+
+[<Test>]
+let ``file end with newline followed by comment, 1649`` () =
+    formatSourceString
+        false
+        """
+#load "Hi.fsx"
+open Something
+
+//// FOO
+[
+    1
+    2
+    3
+]
+
+//// The end
+"""
+        { config with
+              SpaceBeforeUppercaseInvocation = true
+              SpaceBeforeClassConstructor = true
+              SpaceBeforeMember = true
+              SpaceBeforeColon = true
+              SpaceBeforeSemicolon = true
+              MultilineBlockBracketsOnSameColumn = true
+              NewlineBetweenTypeDefinitionAndMembers = true
+              KeepIfThenInSameLine = true
+              AlignFunctionSignatureToIndentation = true
+              AlternativeLongMemberDefinitions = true
+              MultiLineLambdaClosingNewline = true
+              KeepIndentInBranch = true }
+    |> prepend newline
+    |> should
+        equal
+        """
+#load "Hi.fsx"
+open Something
+
+//// FOO
+[ 1 ; 2 ; 3 ]
+
```

Comment:
This is the reason this PR exists in the first place.
Both the newline and comment trivia are now correctly assigned to the same node.

Previously, the newline was assigned to the anonymous module and the comment to the list.
No trivia can now be assigned to the anon module anymore.